controlplane/kubeadm/internal/workload_cluster_test.go

```diff
@@ -227,3 +231,88 @@ func getProxyImageInfo(ctx context.Context, client ctrlclient.Client) (string, e
 	}
 	return container.Image, nil
 }
+
+func TestWorkload_EtcdIsHealthy(t *testing.T) {
+	workload := &Workload{
+		Client: &fakeClient{
+			get: map[string]interface{}{
+				"kube-system/etcd-test-1": etcdPod("etcd-test-1", withReadyOption),
+				"kube-system/etcd-test-2": etcdPod("etcd-test-2", withReadyOption),
+				"kube-system/etcd-test-3": etcdPod("etcd-test-3", withReadyOption),
+				"kube-system/etcd-test-4": etcdPod("etcd-test-4"),
+			},
+			list: &corev1.NodeList{
+				Items: []corev1.Node{
+					nodeNamed("test-1", withProviderID("my-provider-id-1")),
+					nodeNamed("test-2", withProviderID("my-provider-id-2")),
+					nodeNamed("test-3", withProviderID("my-provider-id-3")),
+					nodeNamed("test-4", withProviderID("my-provider-id-4")),
+				},
+			},
+		},
+		etcdClientGenerator: &fakeEtcdClientGenerator{
```

Comment:
this isn't what i want it to look like, but i'm working with the existing structs in order to finish this up